typedmodels/models.py

```diff
@@ -96,9 +96,12 @@ def do_related_class(self, other, cls):
                         old_do_related_class(other, cls)
                         cls._meta.model_name = base_class_name.lower()
                     field.do_related_class = types.MethodType(do_related_class, field)
-                if isinstance(field, models.fields.related.RelatedField) and isinstance(field.rel.to, TypedModel) and field.rel.to.base_class:
-                    field.rel.limit_choices_to['type__in'] = field.rel.to._typedmodels_subtypes
-                    field.rel.to = field.rel.to.base_class
+                if isinstance(field, models.fields.related.RelatedField):
+                    # `rel.to` is required for Django <= 1.8, `remote_field.model` is required for Django >= 2.0.
+                    remote_field_model = field.rel.to if hasattr(field, 'rel') else field.remote_field.model
+                    if isinstance(remote_field_model, TypedModel) and remote_field_model.base_class:
+                        field.rel.limit_choices_to['type__in'] = remote_field_model._typedmodels_subtypes
+                        remote_field_model = remote_field_model.base_class
```

Comment:
This line doesn't do a lot.

The previous code actually _sets_ `field.rel.to`. I guess this was in case you make a `ForeignKey` to a typed proxy model, it changes it to point to the base model.

Sounds too magic to me, what was I thinking? 😄 Maybe just remove this line?